cache/manager.go

```diff
@@ -75,16 +77,32 @@ type cacheManager struct {
 	records map[string]*cacheRecord
 	mu      sync.Mutex
 	ManagerOpt
-	md *metadata.Store
+	store *metadata.Store
 
 	muPrune sync.Mutex // make sure parallel prune is not allowed so there will not be inconsistent results
 	unlazyG flightcontrol.Group
 }
 
 func NewManager(opt ManagerOpt) (Manager, error) {
+	if err := MigrateV2(
+		context.TODO(),
+		filepath.Join(opt.MetadataStoreRoot, "metadata.db"),
+		filepath.Join(opt.MetadataStoreRoot, "metadata_v2.db"),
```

Comment:
Not sure about this change. Why would the cache database need to have a constant name from the package standpoint?

Reply:
It doesn't make sense to me that you can't create two instances of manager with metadata.db in same dir. It is some internal structure limitation that leaks to the public API. I'm ok with for example just adding `MetadataDBV1` or smth in manageropt if you want to move migration here. It is bit weird though that all other stores are initialized and passed in as interface, and these are now passed as strings.